usl_models/usl_models/flood_ml/model_params.py

```diff
@@ -23,7 +24,7 @@ class FloodModelParams:
     # Keras parameters.
     # Keras models support both Optimizer objects and (valid) string names.
     # It is the user's responsibility to pass in a valid optimizer value.
-    optimizer: tf.keras.Optimizer | str = "adam"
+    optimizer: Any = dataclasses.field(default_factory=lambda: tf.keras.optimizers.Adam(learning_rate=1e-3))
```

Comment:
Will this create a default Optimizer that's passed by reference to all instances if it's not overridden?

I switched to the string value in this [commit](https://github.com/UrbanSystemsLab/climateiq-cnn/pull/93/commits/62c5036ef0a2018337185f123204991e5f4bce25) to avoid possible pass-by-reference issues where the same optimizer gets shared across multiple models, which can cause downstream failures.

Reply:
It seems limiting to not be able to specify the parameters of Adam (learning rate etc). Using the default factor avoids these issues as long as the client does not share the Params across multiple models.

Reply:
The default value doesn't limit anything? It's just there if the user doesn't provide their own optimizer, in which they can specify whatever parameters they want. I have the string as a default to prevent any accidental pass-by-reference issues when the user _doesn't_ specify an optimizer. But your optimizer is equivalent to mine.